File: spark-event-gateway-srv/src/main/java/fr/an/spark/gateway/dto/AccumulableInfo.java
package fr.an.spark.gateway.dto;

import org.springframework.lang.Nullable;

import com.fasterxml.jackson.annotation.JsonProperty;

public class AccumulableInfo {

    @JsonProperty("ID")
    public int id;

    @JsonProperty("Name")
    public String name;

    @JsonProperty("Update")
    public long update;

    @JsonProperty("Value")
    public Long value;

    @JsonProperty("Internal")
    public boolean internal;

    @JsonProperty("Count Failed Values")
    public boolean countFailedValues;

    @JsonProperty("Metadata")
    @Nullable
    // public Map<String,Object> metadata;
    // public String metadata;
    public Object metadata;

}

File: spark-events-ui/src/app/model/AccumulableInfo.ts

export class AccumulableInfo {

  // @JsonProperty("ID")
  readonly id: number

  // @JsonProperty("Name")
  readonly name: string;

  // @JsonProperty("Update")
  update: number;

  // @JsonProperty("Value")
  value: number;

  // @JsonProperty("Internal")
  internal: boolean;

  // @JsonProperty("Count Failed Values")
  countFailedValues: boolean;

  // @JsonProperty("Metadata")
  metadata: {}|null;

  // --------------------------------------------------------------------------

  constructor(id: number,
      name: string,
      update: number,
      value: number,
      internal: boolean,
      countFailedValues: boolean,
      metadata: {}|null) {
    this.id = id,
    this.name = name,
    this.update = update;
    this.value = value;
    this.internal = internal;
    this.countFailedValues = countFailedValues;
    this.metadata = metadata;
  }

  static fromJson(src: any): AccumulableInfo {
    let id = <number> src['ID'];
    let name = <string> src['Name'];
    let update = <number> src['Update'];
    let value = <number> src['Value'];
    let internal= <boolean>src['Internal'];
    let countFailedValues = <boolean> src['Count Failed Values'];
    let metadata = src['Metadata'];
    return new AccumulableInfo(id, name, update, value, internal, countFailedValues, metadata);
  }
  
  static fromJsonArray(src: any[]): AccumulableInfo[] {
      return src.map(x => AccumulableInfo.fromJson(x));
  }

}
